.github/workflows/preview-env.yml

```diff
@@ -0,0 +1,46 @@
+name: preview-env-setup-update
+
+on:
+  pull_request:
+    types: [assigned, opened, reopened, synchronize]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        with:
+          submodules: true
+          fetch-depth: 0
+
+      - name: Setup Hugo
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.74.2'
+
+      - name: Build Previews
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        run: |
+          rm -rf ./dist
+          hugo -F -s site --baseURL http://mattermost-dev-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/${{ github.event.number }}/ --destination ../dist/html
+
+      - uses: shallwefootball/s3-upload-action@master
+        name: Upload Preview Env
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        with:
+          aws_key_id: ${{ secrets.AWS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws_bucket: ${{ secrets.AWS_BUCKET }}
+          source_dir: ./dist/html
+          destination_dir: ${{ github.event.number }}
+
+      - name: Add comment to PR
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        uses: peter-evans/create-or-update-comment@v1
```

Comment:
Not sure I like this from a sec perspective.

Reply:
we can review the tagged code and evaluate it. I already did a look and did not see anything that is questinable